src/functions/function-command/run-function.ts

```diff
@@ -0,0 +1,15 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import { knExecutor } from '../../cli/execute';
+import { FuncAPI } from '../../cli/func-api';
+import { FunctionNode } from '../function-tree-view/functionsTreeItem';
+
+export async function runFunction(context?: FunctionNode): Promise<void> {
+  if (!context) {
+    return null;
+  }
+  await knExecutor.executeInTerminal(FuncAPI.runFunc(context.contextPath.fsPath));
```

Comment:
Should we add a guard for contextPath or it contains a valid value in all cases?

Reply:
yes, it will contain the valid value I will test it on windows.